### DISCUSSION TITLE
Rename "emf" to "ecore"?

### DISCUSSION THREAD

**Discussion by alessiostalla** [Strumenta/kolasu#125] [Ideas]
EMF is specifically related to Eclipse (Eclipse Modeling Framework) while Ecore is more generally the model definition library.
Indeed we use ecore.js in Tylasu and pyecore in Pylasu, and these are named "ecore", not "emf".

**Comment by ftomassetti** [comment]
My understanding is that Ecore is how you define metamodels (e.g., our AST classes), and EMF is the overall framework that include the part for defining models (e.g., our AST instances, in other words, the data that we obtain from parsing).

@loradd is probably the one who can confirm or correct this

**Reply by loradd** [reply]
Yes.

Ecore is a (Java-flavoured) OMG EMOF implementation developed as part of EMF. In this last one, Ecore is used both for user and internal metamodels, e.g. GenModel. Given an Ecore metamodel and a GenModel specification (instance), EMF generates code for manipulating instances of the given metamodel. EMF also includes support for registering languages, e.g. editors, etc, within Eclipse.

I am not sure, but I suppose Ecore stands for something like _Eclipse MOF Core_...

What about using _MOF_ referring to the OMG standard?

**Reply by alessiostalla** [reply]
I was thinking that EMF is too Eclipse specific, but if Ecore is just a part of EMF, I think there's no problem in continuing to use "emf".
